### PR TITLE
DCOS-28735: Adjust Jest setup to work with Typescript

### PR DESCRIPTION
**Pull request by nLight**
While working on https://jira.mesosphere.com/browse/DCOS-21930 I spend days trying to fix our Jest setup. 

First, I had to enable Typescript processing. When that was done it turned out that Typescript doesn't like when we use `setMock`, so I removed all usages that don't make sense and replaced other with `mock` which is a recommended way. 

`unMockStores` uses `setMock` under the hood and is going to be removed. First, it doesn't in fact unmock. It actually _sets_ a mock and mocks stores with an empty object. `¯\_(ツ)_/¯ `

**Dependencies**
- Please merge this first: https://github.com/mesosphere/dcos-ui-plugins-private/pull/794

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

Closes DCOS-28735

### CODE REVIEW

**Comment by mesosphere-mergebot-dev** [comment]
This repo has @mesosphere-mergebot integration. You can interact with the following commands.
```
@mesosphere-mergebot ship-it  
@mesosphere-mergebot test  
@mesosphere-mergebot integrate  
```

File: jest/preprocessor.js [hunk 1]
```diff
@@ -21,8 +23,16 @@ module.exports = {
       if (isJISON) {
         src = new jison.Generator(src).generate();
       }
+      if (filename.endsWith(".ts") || filename.endsWith(".tsx")) {
```
Comment (DanielMSchmidt) [inline jest/preprocessor.js]: Is there anything that holds us back from using a separate transform statement? [It should be possible like this](https://github.com/mesosphere/mockserver/blob/master/package.json#L58).

For me this is a lot cleaner and I would like to move away from this file as it has some more complexity

Comment (DanielMSchmidt) [reply jest/preprocessor.js]: If it's about our babel file not being picked up, they have docs on that: https://github.com/kulshekhar/ts-jest#using-a-custom-babel-config

File: jest/setupEnv.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 require("babel-polyfill");
+require("reflect-metadata");
```
Comment (DanielMSchmidt) [inline jest/setupEnv.js]: This is a fragment of the plugins POC, right?

Comment (nLight) [reply jest/setupEnv.js]: It actually should be there already since we're using inversify.

File: jest/preprocessor.js (hunk 1 above)
Comment (nLight) [reply jest/preprocessor.js]: As discussed: I decided to extract jest config refactoring from this piece of work.

**Pull request merged**